tensorflow-tools/src/main/java/org/tensorflow/tools/ndarray/StdArrays.java

```diff
@@ -0,0 +1,1476 @@
+   *
+   * @param array 2D array
+   * @return shape of the array
+   */
+  public static Shape shapeOf(boolean[][] array) {
+    return Shape.of(computeShape(array, new long[2]));
+  }
+
+  /**
+   * Compute the shape of a 3-dimensions boolean array.
+   *
+   * @param array 3D array
+   * @return shape of the array
+   */
+  public static Shape shapeOf(boolean[][][] array) {
+    return Shape.of(computeShape(array, new long[3]));
+  }
+
+  /**
+   * Compute the shape of a 4-dimensions boolean array.
+   *
+   * @param array 4D array
+   * @return shape of the array
+   */
+  public static Shape shapeOf(boolean[][][][] array) {
+    return Shape.of(computeShape(array, new long[4]));
+  }
+
+  /**
+   * Compute the shape of a 5-dimensions boolean array.
+   *
+   * @param array 5D array
+   * @return shape of the array
+   */
+  public static Shape shapeOf(boolean[][][][][] array) {
+    return Shape.of(computeShape(array, new long[5]));
+  }
+
+  /**
+   * Compute the shape of a 6-dimensions boolean array.
+   *
+   * @param array 6D array
+   * @return shape of the array
+   */
+  public static Shape shapeOf(boolean[][][][][][] array) {
+    return Shape.of(computeShape(array, new long[6]));
+  }
+
+  /**
+   * Compute the shape of a single-dimension object array.
+   *
+   * @param array 1D array
+   * @return shape of the array
+   */
+  public static <T> Shape shapeOf(T[] array) {
+    return Shape.of(array.length);
+  }
+
+  /**
+   * Compute the shape of a 3-dimensions object array.
+   *
+   * @param array 2D array
+   * @return shape of the array
+   */
+  public static <T> Shape shapeOf(T[][] array) {
+    return Shape.of(computeShape(array, new long[2]));
+  }
+
+  /**
+   * Compute the shape of a 3-dimensions object array.
+   *
+   * @param array 3D array
+   * @return shape of the array
+   */
+  public static <T> Shape shapeOf(T[][][] array) {
+    return Shape.of(computeShape(array, new long[3]));
+  }
+
+  /**
+   * Compute the shape of a 4-dimensions object array.
+   *
+   * @param array 4D array
+   * @return shape of the array
+   */
+  public static <T> Shape shapeOf(T[][][][] array) {
+    return Shape.of(computeShape(array, new long[4]));
+  }
+
+  /**
+   * Compute the shape of a 5-dimensions object array.
+   *
+   * @param array 5D array
+   * @return shape of the array
+   */
+  public static <T> Shape shapeOf(T[][][][][] array) {
+    return Shape.of(computeShape(array, new long[5]));
+  }
+
+  /**
+   * Compute the shape of a 6-dimensions object array.
+   *
+   * @param array 6D array
+   * @return shape of the array
+   */
+  public static <T> Shape shapeOf(T[][][][][][] array) {
+    return Shape.of(computeShape(array, new long[6]));
+  }
+
+  private static void dimSize(int arrayLength, long[] shape, int dimIdx) {
+    if (shape[dimIdx] == 0) {
+      shape[dimIdx] = arrayLength;
+    } else if (shape[dimIdx] != arrayLength) {
+      shape[dimIdx] = Shape.UNKNOWN_SIZE;
```

Comment:
Is this the behaviour we want for ragged tensors? The ndarray machinery is mostly predicated on the ndarray being a hyperrectangle, is everything robust to getting a `Shape.UNKNOWN_SIZE`?

Reply:
It's in preparation for. Right now, only dense `NdArray`s can be instantiated by TF Tools. Plus, providing a "ragged standard array" to initialize a dense one will fail gracefully, like it is [tested here](https://github.com/karllessard/tensorflow-java/blob/c9cad150cbc245edb92057b9f7e941750080e432/tensorflow-tools/src/test/java/org/tensorflow/tools/ndarray/StdArraysTest.java#L76).

So I think it is safe to support properly unknown dim sizes for standard arrays.